src/common/file_location.c

```diff
@@ -137,58 +157,122 @@ char *dt_loc_find_install_dir(const char *suffix, const char *searchname)
+  darktable.tmpdir = dt_loc_init_generic(tmpdir, NULL, g_get_tmp_dir());
+  dt_check_opendir("darktable.tmpdir", darktable.tmpdir, FALSE);
   if(darktable.tmpdir == NULL) return 1;
   return 0;
 }
 
 void dt_loc_init_user_cache_dir(const char *cachedir)
 {
   char *default_cache_dir = g_build_filename(g_get_user_cache_dir(), "darktable", NULL);
-  darktable.cachedir = dt_loc_init_generic(cachedir, default_cache_dir);
+  darktable.cachedir = dt_loc_init_generic(cachedir, NULL, default_cache_dir);
+  dt_check_opendir("darktable.cachedir", darktable.cachedir, TRUE);
   g_free(default_cache_dir);
 }
 
-void dt_loc_init_plugindir(const char *plugindir)
+void dt_loc_init_plugindir(const char* application_directory, const char *plugindir)
 {
 #if defined(__APPLE__) || defined(_WIN32)
   char *suffix = g_build_filename("lib", "darktable", NULL);
   char *directory = dt_loc_find_install_dir(suffix, darktable.progname);
   g_free(suffix);
-  darktable.plugindir = dt_loc_init_generic(plugindir, directory ? directory : DARKTABLE_LIBDIR);
+  darktable.plugindir = dt_loc_init_generic(plugindir, application_directory, directory ? directory : DARKTABLE_LIBDIR);
+  dt_check_opendir("darktable.plugindir", darktable.plugindir, TRUE);
   g_free(directory);
 #else
-  darktable.plugindir = dt_loc_init_generic(plugindir, DARKTABLE_LIBDIR);
+  darktable.plugindir =  dt_loc_init_generic(plugindir, application_directory, DARKTABLE_LIBDIR);
+  dt_check_opendir("darktable.plugindir", darktable.plugindir, TRUE);
 #endif
 }
 
-void dt_loc_init_localedir(const char *localedir)
+void dt_check_opendir(const char* text, const char* directory, gboolean exit_on_error)
+{
+  if (!directory) {
+    printf("directory for %s has not been set.\n", text);
+    exit(EXIT_FAILURE);
+  } 
+
+  DIR* dir = opendir(directory);
+  if (dir) {
+    dt_print(DT_DEBUG_DEV, "%s: %s\n", text, directory);
+    closedir(dir);
+  } 
+  else 
+  {
+    if ( errno == EACCES ) 
+    {
+      printf("Permission denied.\n");
+    } 
+    else if ( errno == ENOENT ) 
+    {
+      printf("%s %s does not exist.\n", text, directory);
+    } 
+    else if ( errno == EBADF ) 
+    {
+      printf("fd is not a valid file descriptor opened for reading.\n");
+    } 
+    else if ( errno == EMFILE ) 
+    {
+      printf("The per-process limit on the number of open file descriptors has been reached.\n");
+    } 
+    else if ( errno == ENFILE ) 
+    {
+      printf("The system-wide limit on the total number of open files has been reached.\n");
+    } 
+    else if ( errno == ENOENT ) 
+    {
+      printf("Directory does not exist, or name is an empty string.\n");
+    }  
+    else if ( errno == ENOMEM ) 
+    {
+      printf("Insufficient memory to complete the operation.\n");
+    }  
+    else if ( errno == ENOTDIR ) 
+    {
+      printf("Name is not a directory.\n");
+    } 
+    else 
+    {
+      printf("opendir() failed for some other reason.\n");
+    }
+
+    if(exit_on_error)
```

Comment:
If this is the controversial exit() I am fine with it. For all errors above there is no way to have dt running properly. Better exit sooner than crashing later.